uwblueprint/social-diversity-for-children
# Implement Volunteer Enrollment

### Notion ticket link



[Volunteer Enrollment](https://www.notion.so/uwblueprintexecs/086c1b564dbf478faa649a8880e34b81?v=9d3f8002a0d1477ba395616dfc082af7&p=50e0e899d4d94ba7a0fc526c688b749a)



### Implementation description

- separated media release, background check forms instead separate components
- confirmation page component, retrieves selected class to generate confirmation card. changed getClass get api to include program info
- for submit/update background check, redirects to document upload page, after successful upload, closing the document upload page will redirect to the next page in the flow
- upon finishing enrollment, a record is added to the VolunteerRegs table



https://user-images.githubusercontent.com/71860803/137991232-d4090311-0acd-4737-97bf-baba888992eb.mp4



### Steps to test

1. Log in, ensure that your user has the volunteer role.
2. Register through the /program-details/[pid] route using one of the program cards
3. Depending on the values of criminalCheckLink and criminalCheckExpired, either a Submit/Update background check will apear
4. When upload criminal record is clicked, should direct to document upload form. After a file is submitted, the X (Close button) should go back to the enrollment flow, skipping to the next page (Media Release)
5. Click through the separate pages
6. After clicking Finish, a new db record in VolunteerRegs should be created



### What should reviewers focus on?

- 

### Checklist
